src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpatialAlignment/SpatialCoordinateSystemManager.cs

```diff
@@ -491,7 +493,7 @@ private bool TryCleanupExistingLocalizationSession(SpatialCoordinateSystemPartic
             {
                 if (currentLocalizationSession != null)
                 {
-                    if (currentLocalizationSession.Participant == participant)
+                    if (currentLocalizationSession.Participant == participant || currentLocalizationSession.CompletionSource.Task.IsCompleted)

```

Comment:
>currentLocalizationSession [](start = 81, length = 26)

is there more context behind this? Could it be that we aren't setting currentLocalizationSEssion to null when we should be?

Reply:
I'll add a comment that this is needed for the stationary localization + world anchor when they're running in the same session

---
In reply to: [384193089](https://github.com/microsoft/MixedReality-SpectatorView/pull/327#discussion_r384193089) [](ancestors = 384193089)